Client/Frontend/Widgets/BottomSheetViewController.swift

```diff
@@ -0,0 +1,248 @@
+        contentView.addGestureRecognizer(panGestureRecognizer)
+
+        contentView.addSubview(handleView)
+        
+        contentView.addSubview(closeButton)
+        
+        closeButton.addTarget(self, action: #selector(closeView), for: .touchUpInside)
+        contentView.isHidden = true
+        
+        makeConstraints()
+    }
+
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        show()
+    }
+    
+    override func viewDidLayoutSubviews() {
+        yPosition = contentView.isHidden ? view.frame.maxY : initialDrawerYPosition
+        
+        updateDrawerViewConstraints()
+    }
+    
+    override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+        super.traitCollectionDidChange(previousTraitCollection)
+        updateDrawerViewConstraints()
+    }
+    
+    // MARK: - Constraints setup
+    
+    private func makeConstraints() {
+        backgroundOverlayView.snp.makeConstraints { make in
+            let bottomInset = parent?.view.safeAreaInsets.bottom ?? 0
+            make.leading.trailing.top.equalToSuperview()
+            make.bottom.equalToSuperview().offset(bottomInset)
+        }
+        
+        handleView.snp.remakeConstraints {
+            $0.width.equalTo(UX.HandleWidth)
+            $0.height.equalTo(UX.HandleHeight)
+
+            $0.centerX.equalTo(contentView)
+            $0.top.equalTo(contentView).offset((UX.HandleMargin - UX.HandleHeight) / 2)
+        }
+        
+        closeButton.snp.makeConstraints {
+            $0.top.equalToSuperview().inset(10)
+            $0.right.equalToSuperview().inset(7)
+            $0.size.equalTo(26)
+        }
+        
+        updateDrawerViewConstraints()
+    }
+    
+    private func updateDrawerViewConstraints() {
+        let allCorners: CACornerMask = [.layerMinXMinYCorner, .layerMaxXMinYCorner,
+                          .layerMinXMaxYCorner, .layerMaxXMaxYCorner]
+        
+        let onlyTopCorners: CACornerMask = [.layerMinXMinYCorner, .layerMaxXMinYCorner]
+        contentView.layer.maskedCorners = showAsPopup ? allCorners : onlyTopCorners
+        
+        handleView.isHidden = showAsPopup
+        
+        // Don't remake constraints if not needed.
+        if yPosition == initialDrawerYPosition { return }
+        
+        contentView.snp.remakeConstraints {
+            if showAsPopup || UIApplication.shared.statusBarOrientation.isLandscape {
+                $0.bottom.equalToSuperview()
+                $0.centerX.equalToSuperview()
+                $0.width.equalTo(maxHorizontalWidth)
+            } else {
+                $0.leading.trailing.equalToSuperview()
+                $0.bottom.equalToSuperview()
+            }
+        }
+    }
+    
+    // MARK: - Animations
+    
+    @objc private func didRecognizePanGesture(_ gestureRecognizer: UIPanGestureRecognizer) {
```

Comment:
Nit: Name is a bit misleading, it implies it is only called once when it is recognized and not continuously as the user pans.